### PR TITLE
[TASK-7461] feat: add wallet header component

### PR DESCRIPTION
**Pull request by kushagrasarathe**
- added wallet header component

## Release Notes

- **New Features**
  - Introduced `WalletHeader` and `HomeHeader` components for enhanced wallet management and navigation.
  - Added `CopyToClipboard` component for easy text copying functionality.
  - New button variant 'yellow' for additional styling options.
  - New icons: `arrow-down-filled` and `wallet` added for improved UI representation.
  - Optional `hideCloseButton` property in modals for customizable close button visibility.

- **Bug Fixes**
  - Reorganized imports and exports for clarity across various components.

- **Chores**
  - Updated Tailwind CSS configuration with new shades of purple and yellow, along with a new button style.

### CODE REVIEW

File: src/components/Global/CopyToClipboard/index.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+const CopyToClipboard = ({ textToCopy, fill, className }: Props) => {
+    const [copied, setCopied] = useState(false)
+
+    const handleCopy = (e: React.MouseEvent<SVGElement>) => {
+        e.stopPropagation()
+        navigator.clipboard.writeText(textToCopy).then(() => {
+            setCopied(true)
+            setTimeout(() => setCopied(false), 2000)
+        })
+    }
```
[the remark this answers is not in the document]
Comment (Hugo0) [reply src/components/Global/CopyToClipboard/index.tsx:20]: Could use toast here

File: src/app/(mobile-ui)/home/page.tsx [hunk 2]
```diff
@@ -52,6 +52,7 @@ export default function Home() {
             <div className="flex w-full flex-row justify-center overflow-hidden p-4">
                 <div className="flex w-[100%] flex-col gap-4 sm:w-[90%] sm:gap-2 md:w-[70%] lg:w-[50%]">
                     <HomeHeader />
+                    <WalletHeader />
                     <div
```
Comment (Hugo0) [inline src/app/(mobile-ui)/home/page.tsx:56]: Thought: The carousel should prob be a component too

File: src/components/Global/Modal/index.tsx [hunk 3]
```diff
@@ -18,6 +18,7 @@ type ModalProps = {
     classNameWrapperDiv?: string
     showPrev?: boolean
     onPrev?: () => void
+    hideCloseButton?: boolean
```
Comment (Hugo0) [inline src/components/Global/Modal/index.tsx]: Q: Why do we need a hide close button option?

File: src/components/Global/WalletHeader/index.tsx [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+
+interface WalletEntryCardProps {
+    wallet: IWallet
+    isActive?: boolean
+    onClick?: () => void
+}
+
+const WalletHeader = () => {
+    const [showModal, setShowModal] = useState(false)
+    const { wallets, selectedWallet, setSelectedWallet, isConnected } = useWallet()
```
Comment (Hugo0) [inline src/components/Global/WalletHeader/index.tsx:27]: Q: not using isConnected?

File: src/components/Global/WalletHeader/index.tsx [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+    const { connector } = useAccount()
+
+    // sort wallets to add active wallet at the top
+    const sortedWallets = useMemo(() => {
+        return [...wallets].sort((a, b) => {
+            if (a.address === selectedWallet?.address) return -1
+            if (b.address === selectedWallet?.address) return 1
+            return 0
+        })
+    }, [wallets, selectedWallet])
```
Comment (Hugo0) [inline src/components/Global/WalletHeader/index.tsx:38]: Issue, blocking: This wallet ordering seems to be wrong - We should show the wallets in the home screen by this order:

left-most: Peanut wallet
2nd: 2nd added external wallet
3rd: 3rd added external wallet 

etc..

The active wallet should be the one selected, but not the left-most one

**Review by Hugo0 — APPROVED** [review]
~Except for wallet ordering, rest~ lgtm!

File: src/components/Global/WalletHeader/index.tsx [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+        setSelectedWallet(wallet)
+        setShowModal(false)
+    }
+
+    return (
+        <div>
+            {/* wallet selector button with current wallet info */}
+            <Button
+                variant="yellow"
+                className="h-auto w-fit min-w-32 bg-yellow-1 bg-opacity-20 py-2"
```
Comment (Hugo0) [inline src/components/Global/WalletHeader/index.tsx]: Q: why opacity 20? where is this coming from?

Comment (kushagrasarathe) [reply src/components/Global/WalletHeader/index.tsx]: yeah this was one of the doubts on DS, cleared it yesterday, updating and pushing the change now

File: src/components/Global/Modal/index.tsx (hunk 3 above)
Comment (kushagrasarathe) [reply src/components/Global/Modal/index.tsx]: was planning to move the modal to look like sheet for mobile screens, figured that's a little lengthy task and not high priority, for now reverted it to old structure, tracking it to and will tackle in another spirint

**Pull request merged**